File: BlockPartyWindowsStore/Screens/SplashScreen.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlockPartyWindowsStore
{
    class SplashScreen : Screen
    {
        TimeSpan timeElapsed;
        readonly TimeSpan duration = TimeSpan.FromSeconds(1);

        Texture2D logoTexture;

        public SplashScreen(ScreenManager screenManager) : base(screenManager) 
        {
            TransitionDuration = TimeSpan.FromSeconds(1);
        }

        public override void LoadContent()
        {
            base.LoadContent();

            logoTexture = ContentManager.Load<Texture2D>("Logo");
        }

        public override void Update(GameTime gameTime)
        {
            base.Update(gameTime);

            if (State == Screen.ScreenState.Active)
            {
                timeElapsed += gameTime.ElapsedGameTime;

                if (timeElapsed >= duration)
                {
                    ScreenManager.LoadScreen(new MainMenuScreen(ScreenManager));
                }
            }
        }

        public override void Draw(Microsoft.Xna.Framework.GameTime gameTime)
        {
            ScreenManager.GraphicsManager.DrawFullscreenSprite(logoTexture, Color.White);

            base.Draw(gameTime);
        }
    }
}

File: BlockPartyWebRole/Client/SplashScreen.ts
/// <reference path="MainMenuScreen.ts" />
/// <reference path="GameScreen.ts" />
/// <reference path="TimeSpan.ts" />

class SplashScreen extends GameScreen {
    private timeElapsed: TimeSpan = TimeSpan.Zero;
    private duration: TimeSpan = TimeSpan.FromSeconds(1);
    private logoImage: HTMLImageElement = new Image();

    constructor(screenManager: ScreenManager) {
        super(screenManager);

        this.TransitionDuration = TimeSpan.FromSeconds(1);
    }

    public LoadContent() {
        this.logoImage.src = "Client/Images/OmegaSplashScreen.png";
    }

    public Update(gameTime: GameTime) {
        super.Update(gameTime);

        if (this.State == ScreenState.Active) {
            this.timeElapsed = new TimeSpan(this.timeElapsed.Milliseconds + gameTime.ElapsedGameTime.Milliseconds);

            if (this.timeElapsed.Milliseconds >= this.duration.Milliseconds) {
                this.ScreenManager.LoadScreen(new MainMenuScreen(this.ScreenManager));
            }
        }
    }

    public Draw(gameTime: GameTime) {
        this.ScreenManager.Game.GraphicsManager.Draw(this.logoImage, new Rectangle(this.ScreenManager.Game.WorldViewport.X, this.ScreenManager.Game.WorldViewport.Y, this.ScreenManager.Game.WorldViewport.Width, this.ScreenManager.Game.WorldViewport.Height), "white");

        super.Draw(gameTime);
    }
}